regulation-worker/internal/delete/api/api.go

```diff
@@ -223,9 +224,12 @@ func (api *APIManager) refreshOAuthToken(destName, workspaceId string, oAuthDeta
 		DestDefName:     destName,
 		EventNamePrefix: "refresh_token",
 	}
-	statusCode, _ := api.OAuth.RefreshToken(refTokenParams)
+	statusCode, refreshResponse := api.OAuth.RefreshToken(refTokenParams)
+	if refreshResponse == nil {
+		return fmt.Errorf("[%v] Failed to refresh token for destination in workspace(%v) & account(%v) due to `nil` response", destName, workspaceId, oAuthDetail.id)
+	}
 	if statusCode != http.StatusOK {
-		return fmt.Errorf("failed to refresh token for destination: %v", destName)
+		return fmt.Errorf("[%v] Failed to refresh token for destination in workspace(%v) & account(%v) with %v", destName, workspaceId, oAuthDetail.id, refreshResponse.Err)
 	}
```

Comment:
we shouldn't skip the `statusCode` check, right?
```
statusCode, refreshResponse := api.OAuth.RefreshToken(refTokenParams)
if statusCode != http.StatusOK {
	var responseErr string
	if refreshResponse != nil {
		responseErr = refreshResponse.Err
	}
	return fmt.Errorf("[%v] Failed to refresh token for destination in workspace(%v) & account(%v) with %v", destName, workspaceId, oAuthDetail.id, responseErr)
}
```

Reply:
Yeah, makes sense! Made the change